src/pytest/test_wall.py

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2018-2020, Michael P. Howard
+# Copyright (c) 2021-2025, Auburn University
+# Part of azplugins, released under the BSD 3-Clause License.
+
+"""Wall potential unit tests."""
+
+import collections
+
+import hoomd
+import hoomd.azplugins
+import numpy
+
+import pytest
+
+PotentialTestCase = collections.namedtuple(
+    "PotentialTestCase",
+    ["potential", "params", "wall", "position", "energy", "force"],
+)
+
+potential_tests = []
+
+# Lennard Jones 9-3
+potential_tests += [
+    # test the calculation of force and potential for plane
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Plane(origin=(0, 0, 0), normal=(0, 0, 1)),
+        (0, 0, 1.5),
+        -0.2558,
+        -0.5718,
+    ),
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Plane(origin=(0, 0, 0), normal=(0, 0, 1)),
+        (0, 0, 3.5),
+        0,
+        0,
+    ),
+    # test the calculation of force and potential for sphere
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Sphere(radius=5, origin=(0, 0, 0), inside=True),
+        (0, 0, 3.5),
+        -0.2558,
+        0.5718,
+    ),
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Sphere(radius=5, origin=(0, 0, 0), inside=True),
+        (0, 0, 2.0),
+        0,
+        0,
+    ),
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Sphere(radius=5, origin=(0, 0, 0), inside=True),
+        (0, 0, 5.5),
+        0,
+        0,
+    ),
+    # test the calculation of force and potential for cylinder
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Cylinder(radius=5, origin=(0, 0, 0), axis=(1, 0, 0), inside=True),
+        (0, 0, 3.5),
+        -0.2558,
+        0.5718,
+    ),
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Cylinder(radius=5, origin=(0, 0, 0), axis=(1, 0, 0), inside=True),
+        (0, 0, 5.5),
+        0,
+        0,
+    ),
+    PotentialTestCase(
+        hoomd.azplugins.wall.LJ93,
+        {"sigma": 1.0, "epsilon": 1.0, "r_cut": 3.0, "r_extrap": 0.0},
+        hoomd.wall.Sphere(radius=5, origin=(0, 0, 0), inside=True),
+        (0, 0, 2.0),
+        0,
+        0,
+    ),
```

Comment:
Since the purpose of these tests is to check the functionality of the evaluator, I think we can remove these and just test the plane. We'll assume the wall potentials work correctly for all geometries.